### PR TITLE
Parameterize Java build number and ensure java is installed

### PR DESCRIPTION
**Pull request by hgontijo**
* Allow java build number to be parameterized
* Ensure installed java version is configured in the enviroment

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 ---
 oracle_jdk_version: 8
 oracle_jdk_update: 74
+oracle_jdk_beta: b02
```
Comment (medzin) [inline defaults/main.yml]: oracle_jdk_build_number perhaps?

File: defaults/main.yml [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+oracle_jdk_rpm_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}-linux-x64.rpm
+oracle_jdk_rpm_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}/jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.rpm
 
 # Debian family specific variables
-oracle_jdk_tar_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.tar.gz
-oracle_jdk_tar_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-b02/{{ oracle_jdk_tar_download_filename }}
+oracle_jdk_tar_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}-linux-x64.tar.gz
+oracle_jdk_tar_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}/jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.rpm
```
Comment (medzin) [inline defaults/main.yml]: For Debian you should download tar.gz package - also use oracle_jdk_tar_download_filename variable here.

File: defaults/main.yml [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 # RedHat family specific variables
-oracle_jdk_rpm_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.rpm
-oracle_jdk_rpm_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-b02/{{ oracle_jdk_rpm_download_filename }}
+oracle_jdk_rpm_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}-linux-x64.rpm
+oracle_jdk_rpm_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}/jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.rpm
 
 # Debian family specific variables
-oracle_jdk_tar_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.tar.gz
-oracle_jdk_tar_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-b02/{{ oracle_jdk_tar_download_filename }}
+oracle_jdk_tar_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}-linux-x64.tar.gz
```
Comment (medzin) [inline defaults/main.yml]: AFAIK filename is without build number.

File: defaults/main.yml [hunk 4]
```diff
@@ -1,11 +1,12 @@
 ---
 oracle_jdk_version: 8
 oracle_jdk_update: 74
+oracle_jdk_build_number: b02
 
 # RedHat family specific variables
-oracle_jdk_rpm_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.rpm
-oracle_jdk_rpm_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-b02/{{ oracle_jdk_rpm_download_filename }}
+oracle_jdk_rpm_download_filename: jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}-linux-x64.rpm
+oracle_jdk_rpm_download_source_url: http://download.oracle.com/otn-pub/java/jdk/{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-{{ oracle_jdk_build_number }}/jdk-{{ oracle_jdk_version }}u{{ oracle_jdk_update }}-linux-x64.rpm
```
Comment (medzin) [inline defaults/main.yml]: Don't double filename creation logic here - use oracle_jdk_rpm_download_filename variable.

File: defaults/main.yml (hunk 3 above)
Comment (hgontijo) [inline defaults/main.yml]: I added it to differentiate multiple build number (e.g. b01, b02) of a same version (e.g. 8u73). If you have an another approach, please let me know.
Without this code, a sequential execution with same version and different build numbers will end up with the very first (build number) installed regardless how many times you execute with different build numbers.

Comment (medzin) [inline defaults/main.yml]: ok let's leave it that way

File: defaults/main.yml (hunk 1 above)
Comment (hgontijo) [inline defaults/main.yml]: fixed.

File: defaults/main.yml (hunk 2 above)
Comment (hgontijo) [inline defaults/main.yml]: Fixed.

File: defaults/main.yml (hunk 4 above)
Comment (hgontijo) [inline defaults/main.yml]: Fixed.

File: defaults/main.yml (hunk 3 above)
Comment (hgontijo) [inline defaults/main.yml]: in fact, I tried installing different build numbers (1.8.0_71-b14, 1.8.0_71-b15) however yum just see both as similar package. In other words, sequential 'yum install' executions of the same version (e.g. 1.8.0_71) and different build number (e.g. b14, b15) does not reinstall the package.

**Pull request merged**